File: save_humanity/humanity.py
import sys

def findMatches(patient, virus):
    if len(virus) > len(patient):
        return 0
    
    foundMatch = False
    
    # try all possible starting positions for the virus
    for startPos in range(len(patient) - len(virus) + 1):
        hadError = False
        allMatch = True
        
        # check if rest of string matches
        for i in range(len(virus)):
            if patient[startPos + i] != virus[i]:
                if not hadError:
                    hadError = True
                else:
                    allMatch = False
                    break
                
        if allMatch:
            if foundMatch:
                sys.stdout.write(' ')
            else:
                foundMatch = True
            sys.stdout.write(str(startPos))
    
    sys.stdout.write('\n')

# read number of samples
N = int(sys.stdin.readline())

for i in range(N):
    # read data
    patient = sys.stdin.readline().strip()
    virus = sys.stdin.readline().strip()
    
    # skip blank line after sample
    sys.stdin.readline()
    
    findMatches(patient, virus)

File: save_humanity/humanity.cc
#include <iostream>
#include <string.h>

using namespace std;

void findMatches(char* patient, char* virus) {
	int patLen = strlen(patient);
	int virLen = strlen(virus);

	if (patLen < virLen) {
		return;
	}

	bool hadError = false, allMatch = true;
	int i, j;

	// try all possible starts for the virus
	for (i = 0; i < patLen - virLen + 1; ++i) {
		hadError = false;
		allMatch = true;

		// try to match virus, allow one error
		for (j = 0; j < virLen; ++j) {
			if (patient[i + j] != virus[j]) {
				if (hadError) {
					allMatch = false;
					break;
				}
				else {
					hadError = true;
				}
			}
		}

		if (allMatch) {
			cout << i << " ";
		}
	}

	cout << endl;
}

int main() {
    int i, n;
    char patient[1000001], virus[100001];

    // read number of patients
    cin >> n;

    for (i = 0; i < n; ++i) {
    	// read strings
        cin >> patient;
        cin >> virus;

        // find matches
        findMatches(patient, virus);
    }

    return 0;
}
